frontend/src/components/ui/adminPortal/SearchBar/shadcn-io/navbar-14/index.tsx

```diff
@@ -0,0 +1,147 @@
+'use client';
+
+import * as React from 'react';
+import { useId, useState, useRef, useEffect } from 'react';
+
+import { data, type Member } from "@/components/dummyData/members";
+import { LayoutGridIcon, FunnelIcon, SearchIcon } from 'lucide-react';
+import { Button } from '@/components/ui/button';
+import { Input } from '@/components/ui/input';
+import { cn } from '@/lib/utils';
+
+export interface Navbar14Props extends React.HTMLAttributes<HTMLElement> {
+  searchPlaceholder?: string;
+  searchValue?: string;
+  onSearchChange?: (value: string) => void;
+  onLayoutClick?: () => void;
+}
+
+export const Navbar14 = React.forwardRef<HTMLElement, Navbar14Props>(
+  (
+    {
+      className,
+      searchPlaceholder = 'Search for members...',
+      onLayoutClick,
```

Comment:
Need to add functionality to switch between gallery and list view (by specifying `onLayoutClick` prop).